micrometer-spring-legacy/src/main/java/io/micrometer/spring/advice/TimedMethodInterceptor.java

```diff
@@ -0,0 +1,88 @@
+package io.micrometer.spring.advice;
+
+import io.micrometer.core.annotation.Timed;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tags;
+import io.micrometer.core.instrument.Timer;
+import io.micrometer.core.instrument.Timer.Builder;
+import org.aopalliance.intercept.MethodInterceptor;
+import org.aopalliance.intercept.MethodInvocation;
+import org.springframework.core.annotation.AnnotatedElementUtils;
+import org.springframework.util.ReflectionUtils;
+
+import java.lang.reflect.Method;
+
+public class TimedMethodInterceptor implements MethodInterceptor {
+    public static final String DEFAULT_METRIC_NAME = "method_timed";
+    private final MeterRegistry registry;
+    private final TimedMetricNameResolver timedMetricNameResolver;
+    private final TimedTagsResolver timedTagsResolver;
+
+    public TimedMethodInterceptor(MeterRegistry registry) {
+        this(registry, invocation ->
+            Tags.of("class", invocation.getMethod().getDeclaringClass().getSimpleName(),
```

Comment:
If however not super-likely - I think it would be better with the fully qualified classname (or another tag for package) - so that we ensure there is no possible clash.

Reply:
I would like to go with another tag for package to not blow up size of class tag. However I don't think someone will put `@Timed` on two different classes with the same name.